tests/allman/issue0578.d.ref

```diff
@@ -0,0 +1,43 @@
+void f()
+{
+    auto t = true ? 1 : 0;
+    auto a = [true ? 1 : 0];
+    auto aa1 = [0: true ? 1 : 0];
+    auto aa2 = [0: true ? (false ? 1 : 2) : 3];
+
+    auto aa3 = [0: true ? false ? 1: 2 : 3];
```

Comment:
this seems to be a part of the issue you said would be fixed in the title, no?

Reply:
Yes, that's a bug in libdparse(*1), when that is fixed it will format correctly.
I added that test so when libdparse is fixed and this fails, someone can confirm that the new formatting is correct, adjust the test and remove the long explanation comment within the test.

\*1 From what I can tell libdparse doesn't recognise nested ternary expressions so the inner one doesn't get parsed by `visit(TernaryExpression ...)`. That doesn't matter for line 6 where when testing the associated array condition for the colon of the inner ternary expression, the `indents.topIs(tok!"]")` matches a parenthesis instead of a `]`.
I have opened [this issue](https://github.com/dlang-community/libdparse/issues/507#issue-1955324369) in libdparse.

Reply:
My bad I misunderstood libdparse, I fix it